### ISSUE TITLE
lombok plugin provokes dependency warning in gradle when used with netflix dgs plugin (similar to issue #442)

### ISSUE THREAD

**Issue by michaeljacksonatvyrl** [freefair/gradle-plugins#478]
Hello, it seems that (similar to issue #442) there's a Gradle dependency warning raised when the Netflix DGS codegen plugin is added to a project already using the freefair Lombok plugin.

**Minimal reproduction steps:**

1. Make test directory and cd to it
2. gradle init (I chose java application, defaults for everything)
3. put the following in root-level build.gradle:
```
plugins {
  id 'io.freefair.lombok' version '6.4.0'
  id 'com.netflix.dgs.codegen' version '5.1.16'
}
```
4. run "gradle jar" or similar to try a build.

**Observed warnings from Gradle**

The warnings observed are similar to (omitting the ones caused by e.g. no schema being present and the like):
```
  - Gradle detected a problem with the following location: '/Users/mjackson/test/build/generated/sources/dgs-codegen'. Reason: Task ':generateEffectiveLombokConfig' uses this output of task ':generateJava' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

**Environment info**
```
mjackson ~/test % gradle --version
------------------------------------------------------------
Gradle 7.3.1
------------------------------------------------------------
Build time:   2021-12-01 15:42:20 UTC
Revision:     2c62cec93e0b15a7d2cd68746f3348796d6d42bd

Kotlin:       1.5.31
Groovy:       3.0.9
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          11.0.12 (Homebrew 11.0.12+0)
OS:           Mac OS X 12.1 aarch64
```
(also tested with a 7.4 gradlew in the project where these warnings were first observed)

Is there a way to either manually specify that task dependency or perhaps instruct lombok to skip the generated sourceset(s) as they don't contain Lombok'd code? 

Thank you! :)

**Comment by larsgrefer** [comment]
I think this happens because the dgs plugin doesn't model its task-dependencies correctly. I'll have a look at their plugin.

https://github.com/Netflix/dgs-codegen/issues/190 might also be related

**Comment by michaeljacksonatvyrl** [comment]
Thank you so much! You rock! :)

**Comment by michaeljacksonatvyrl** [comment]
btw, verified this is fixed in the latest from netflix dgs, 5.1.17, that includes your PR. thank you again! :)
